rfc/sqlDesignDocs.md

```diff
@@ -0,0 +1,84 @@
+# Sql Design Docs
+
+## Overview
+
+Currently Fili only supports Druid for performing aggregations/grouping/post aggregations. Fili's processing pipeline is capable of sending queries to different backends, but lacks 
+
+1. the translation (serialization) from `DruidQuery` to other query formats. 
+2. the ability to manually build up a `ResultSet` and pass it through the `ResponseProcessor` instead of passing Druid's son responses
+
+This RFC aims to solve the first issue - lack of translation to another query model by using Calcite to build up SQL queries and processing the results into a Druid format ( `GroupByQuery` response) which Fili is able to understand.
+
+## Goals
+
+Allow customers to query Fili's API and make a `DruidAggregationQuery`  which will be processed by a SQL backed datasource instead of Druid.
+
+Customers will be able to configure their connection to a SQL database similar to Druid through the `applicationConfig.properties` and build a table in Fili specifically backed by SQL. They can then depend on the `fili`  and `fili-sql`module as opposed for configuring their application with SQL support.
+
+## Implementation
+
+1. Create a `fili-sql` module.
+
+2. Define a `SqlBackedClient` 
+
+   ```java
+   Future<JsonNode> executeQuery(
+     DruidQuery<?> druidQuery,
+     SuccessCallback successCallback,
+     FailureCallback failureCallback
+   ); //SqlBackedClient.java
+   ```
+
+   This method accepts and translates a `DruidQuery` to SQL, executes it, reads the results, and builds Druid json results.
+
+3. Define a `DruidQueryToSqlConverter` so that we can query the backend
+
+   1. Define a `SqlTimeConverter` which handles grouping/filtering on intervals. The default behaviour for grouping is to explode the timegrain out into the equivalent DateTime parts for grouping, then reassemble a DateTime when processing the results.
+
+      ```java
+      AllGranularity.INSTANCE -> ()
+      DefaultTimeGrain.YEAR   -> (YEAR)
+      DefaultTimeGrain.MONTH  -> (YEAR, MONTH)
+      DefaultTimeGrain.WEEK   -> (YEAR, WEEK)
+      DefaultTimeGrain.DAY    -> (YEAR, DAYOFYEAR)
+      DefaultTimeGrain.HOUR   -> (YEAR, DAYOFYEAR, HOUR)
+      DefaultTimeGrain.MINUTE -> (YEAR, DAYOFYEAR, HOUR, MINUTE)
+      ```
+
+   2. Define an `ApiToFieldMapper` to handle converting from API names to physical names.
```

Comment:
This may have been unnecessarily abstracted then because my implementation of `ApiToFieldMapper` is using the physical table. 

This is just a helper class that seemed useful to me, but it might not have been useful to put it in the RFC